components/screens/browser-api/sign-tx/SignTx.tsx

```diff
@@ -0,0 +1,117 @@
+import { ApiResponse, SignTxPayload } from "@/api/message";
+import { ApiExtensionUtils } from "@/api/extension";
+import { NetworkType } from "@/contexts/SettingsContext.tsx";
+import { IWallet } from "@/lib/wallet/interface";
+import { useBoolean } from "usehooks-ts";
+import useWalletManager from "@/hooks/useWalletManager";
+import useRpcClientStateful from "@/hooks/useRpcClientStateful";
+import { Transaction } from "@/wasm/core/kaspa";
+
+type SignTxProps = {
+  wallet: IWallet;
+  networkId: NetworkType;
+  requestId: string;
+  payload: SignTxPayload;
+};
+
+export default function SignTx({
+  wallet,
+  networkId,
+  requestId,
+  payload,
+}: SignTxProps) {
+  const { value: isLoading, toggle: toggleLoading } = useBoolean(false);
```

Comment:
I have seen this pattern a few times maybe we should have a `useLoader()` custom hook.

Reply:
This pattern only happens in `SignTx` and `SignAndBroadcast` for temporarily rough ui. I would improve it when the UI is ready.